simpleclient_spring_web/src/main/java/io/prometheus/client/spring/web/PrometheusTimeMethods.java

```diff
@@ -0,0 +1,46 @@
+package io.prometheus.client.spring.web;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Enable Spring-AOP-based automated method timing for the annotated method or all methods in an annotated class.
+ * The timings will be recorded in a Histogram with the default metric name of {@code prometheus_method_timing},
+ * which can be overridden by providing a {@code value} parameter in the annotation, and optionally a {@code help}
+ * parameter and a single label of {@code signature} whose values will be {@code ClassOrInterfaceName.methodName(..)}.
+ *
+ * To properly work, {@link EnablePrometheusTiming} must be specified somewhere in your application configuration.
+ *
+ *  <pre><code>
+ * {@literal @}PrometheusTimeMethods("my_app_timer_seconds")
```

Comment:
So were they seeing performance issues in the application itself, due to Prometheus overhead? I guess I was assuming that the metrics overhead (for metrics instances themselves) was pretty negligible compared to the methods being profiled.

I think my main concern is that the primary point of the annotation approach is ease of use, and I'm not sure we do much more than annoy users and force more typing if they really want to annotate every method of a class. If they do want to, they'll still just use the annotation on every method, which just forces them to type more and come up with more help phrases and metric names, and then make sure they maintain the code any time new methods are added in case they want those methods profiled too.

Honestly, I'm not firmly set on allowing class annotations, but to me it feels like the better experience, as long as it goes with the warning that "hey, use metrics sparingly," which seems like it applies regardless of whether the metrics were created with annotations, or by adding all the requisite code themselves.

Thoughts?